server/server.js

```diff
@@ -113,10 +135,17 @@ function handleRequest(request, response) {
         var headers = { Location: "boards/" + encodeURIComponent(boardName) };
         response.writeHead(301, headers);
         response.end();
-      } else if (parts.length === 2 && request.url.indexOf(".") === -1) {
+      } else if (parts.length === 2 && parsedUrl.pathname.indexOf(".") === -1) {
         validateBoardName(parts[1]);
         // If there is no dot and no directory, parts[1] is the board name
-        boardTemplate.serve(request, response);
+        try {
+          userHasPermission(parsedUrl);
+          // User has permission so we can proceed
+          boardTemplate.serve(request, response);
+        } catch (error) {
+          response.writeHead(403, { 'Content-Type': 'text/plain' });
+          response.end("Forbidden\n" + error);
+        }
```

Comment:
Why did you duplicate the permission check into every part of the router ?  Can we have a single check at the top ?

Reply:
This is so that resources are still returned when they are requested, as a resource won't have a token attached to the request. The permission check is in every section except for when resources are requested.